src/handler/sign_tx.c

```diff
@@ -21,6 +21,14 @@
 #include "../token/token_parser.h"
 #include "../token/types.h"
 
+/**
+ * Helper to clean global context of any sign data
+ */
+void clean_globals() {
+    explicit_bzero(&G_context, sizeof(G_context));
+    explicit_bzero(&G_token_symbols, sizeof(G_token_symbols));
+}
```

Comment:
DRY: Don't Repeat Yourself
This was being required on many places where we need to cleanup the state.